src/commands/config/generate.ts

```diff
@@ -121,14 +121,31 @@ export default class ConfigGenerateCmd extends Command {
 
 	public static authenticated = true;
 
-	public async getApplication(balena: BalenaSDK, fleet: string) {
+	private static readonly getApplicationOptions = {
+		$select: 'slug',
+		$expand: {
+			is_for__device_type: { $select: 'slug' },
+		},
+	} as const;
+
+	public async getApplication(
+		balena: BalenaSDK,
+		fleet: string,
+	): Promise<
+		NonNullable<
+			Pine.OptionsToResponse<
+				Application['Read'],
+				typeof ConfigGenerateCmd.getApplicationOptions,
+				string
+			>
+		>
+	> {
 		const { getApplication } = await import('../../utils/sdk');
-		return await getApplication(balena, fleet, {
-			$select: 'slug',
-			$expand: {
-				is_for__device_type: { $select: 'slug' },
-			},
-		});
+		return await getApplication(
+			balena,
+			fleet,
+			ConfigGenerateCmd.getApplicationOptions,
+		);
 	}
```

Comment:
NIT: So that we can be sure that the object is only allocated when needed & GCed after that.
```suggestion
	public async getApplication(
		balena: BalenaSDK,
		fleet: string,
	) {
		const { getApplication } = await import('../../utils/sdk');
		const options = {
			$select: 'slug',
			$expand: {
				is_for__device_type: { $select: 'slug' },
			},
		}
		const result: NonNullable<
			Pine.OptionsToResponse<
				Application['Read'],
				typeof options,
				string
			>
		> = await getApplication(
			balena,
			fleet,
			options,
		);
		return result;
}
```

Reply:
@thgreasi This does not work as for some particular cases (such as this one) we need the response type defined explicitly

Reply:
@thgreasi  It can't be inlined as it is used in two places - the explicit function type return (otherwise the generated typings need a inner dep to pinejs-client-core -> abstract-sql-to-typescript to be able to be generated)

I don't mind moving it to top level, that should be fine